# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/TextEditor.h
# Commit: 7b4beaa4fab1
# Full Hash: 7b4beaa4fab16011913fea3582f0d89f22890c78
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2025-03-13 10:29:34
# Regressor Bug: 1717155
# File Overlap Count: 1
# Description:
#   Bug 1717155 - part 1: Make `TextEditor` use `GetTextNode()` more r=m_kato
#   
#   The anonymous `<div>` of `TextEditor` has only one `Text` as its first child.
#   Therefore, we can make `TextEditor` simpler to access the `Text` node.
#   
# ==============================================================================

diff -r 267514795e90 -r 7b4beaa4fab1 editor/libeditor/TextEditor.h
--- a/editor/libeditor/TextEditor.h	Thu Mar 13 04:48:38 2025 +0000
+++ b/editor/libeditor/TextEditor.h	Thu Mar 13 05:25:45 2025 +0000
@@ -259,6 +259,10 @@
     }
   }
 
+  /**
+   * Return the `Text` node in the anonymous <div>.  Note that the anonymous
+   * <div> can have only one `Text` for the storage of the value of this editor.
+   */
   dom::Text* GetTextNode() {
     MOZ_DIAGNOSTIC_ASSERT(GetRoot());
     MOZ_DIAGNOSTIC_ASSERT(GetRoot()->GetFirstChild());
